test dataset are not initialized

After the sudden move of all tests datasets definition from pw to pwem, the following command fails:

```
azazello@debian:~/soft/scipion3$ scipion3 tests --grep eman --run
Scipion v3.0.7 - Eugenius
Running tests....
>>>> xmipp3
>>>> atlas
>>>> bamfordlab
>>>> bsoft
>>>> ccp4
>>>> cryoef
>>>> cryodrgn
>>>> emxlib
>>>> fsc3d
>>>> gautomatch
>>>> imagic
>>>> localrec
>>>> locscale
>>>> motioncorr
>>>> phenix
>>>> resmap
>>>> sidesplitter
>>>> sphire
>>>> spider
>>>> topaz
>>>> chimera
>>>> eman2
python3 /home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/apps/pw_run_tests.py    eman2.tests.test_protocols_eman.TestEmanTiltValidate
Running tests....
Creating project at:  /home/azazello/ScipionUserData/projects/TestEmanTiltValidate/project.sqlite
[   FAILED ] TestEmanTiltValidate.setUpClass

Traceback (most recent call last):
  File "/home/azazello/soft/miniconda3/envs/.scipion3env/lib/python3.8/unittest/suite.py", line 163, in _handleClassSetUp                                                  
    setUpClass()                                                                                                                                                           
  File "/home/azazello/soft/scipion-em-plugins/scipion-em-eman2/eman2/tests/test_protocols_eman.py", line 263, in setUpClass                                               
    cls.dataset = DataSet.getDataSet('eman')                                                                                                                               
  File "/home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/tests/tests.py", line 57, in getDataSet                                                                 
    assert name in cls._datasetDict, "Dataset: %s dataset doesn't exist." % name                                                                                           
AssertionError: Dataset: eman dataset doesn't exist.
``` 

Individual tests still work though.

I should have warned you. Datasets were moved to scipion-em and we need a
small change in scipion-app. Git pull of scipion app should fix it

El dom., 14 mar. 2021 16:13, Grigory Sharov ***@***.***>
escribió:

> After the sudden move of all tests datasets definition from pw to pwem,
> the following command fails:
>
> ***@***.***:~/soft/scipion3$ scipion3 tests --grep eman --run
> Scipion v3.0.7 - Eugenius
> Running tests....
> >>>> xmipp3
> >>>> atlas
> >>>> bamfordlab
> >>>> bsoft
> >>>> ccp4
> >>>> cryoef
> >>>> cryodrgn
> >>>> emxlib
> >>>> fsc3d
> >>>> gautomatch
> >>>> imagic
> >>>> localrec
> >>>> locscale
> >>>> motioncorr
> >>>> phenix
> >>>> resmap
> >>>> sidesplitter
> >>>> sphire
> >>>> spider
> >>>> topaz
> >>>> chimera
> >>>> eman2
> python3 /home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/apps/pw_run_tests.py    eman2.tests.test_protocols_eman.TestEmanTiltValidate
> Running tests....
> Creating project at:  /home/azazello/ScipionUserData/projects/TestEmanTiltValidate/project.sqlite
> [   FAILED ] TestEmanTiltValidate.setUpClass
>
> Traceback (most recent call last):
>   File "/home/azazello/soft/miniconda3/envs/.scipion3env/lib/python3.8/unittest/suite.py", line 163, in _handleClassSetUp
>     setUpClass()
>   File "/home/azazello/soft/scipion-em-plugins/scipion-em-eman2/eman2/tests/test_protocols_eman.py", line 263, in setUpClass
>     cls.dataset = DataSet.getDataSet('eman')
>   File "/home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/tests/tests.py", line 57, in getDataSet
>     assert name in cls._datasetDict, "Dataset: %s dataset doesn't exist." % name
> AssertionError: Dataset: eman dataset doesn't exist.
>
> Individual tests still work though.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipion-em/scipion-em/issues/200>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF7ZYP4XI3L3LLJZUQHWQTTDTHAFANCNFSM4ZFCBUJA>
> .
>

I have done git pull in all repos before creating this issue.

Best regards,
Grigory

--------------------------------------------------------------------------------
Grigory Sharov, Ph.D.

MRC Laboratory of Molecular Biology,
Francis Crick Avenue,
Cambridge Biomedical Campus,
Cambridge CB2 0QH, UK.
tel. +44 (0) 1223 267228 <+44%201223%20267228>
e-mail: ***@***.***


On Sun, Mar 14, 2021 at 6:49 PM Pablo Conesa ***@***.***>
wrote:

> I should have warned you. Datasets were moved to scipion-em and we need a
> small change in scipion-app. Git pull of scipion app should fix it
>
> El dom., 14 mar. 2021 16:13, Grigory Sharov ***@***.***>
> escribió:
>
> > After the sudden move of all tests datasets definition from pw to pwem,
> > the following command fails:
> >
> > ***@***.***:~/soft/scipion3$ scipion3 tests --grep eman --run
> > Scipion v3.0.7 - Eugenius
> > Running tests....
> > >>>> xmipp3
> > >>>> atlas
> > >>>> bamfordlab
> > >>>> bsoft
> > >>>> ccp4
> > >>>> cryoef
> > >>>> cryodrgn
> > >>>> emxlib
> > >>>> fsc3d
> > >>>> gautomatch
> > >>>> imagic
> > >>>> localrec
> > >>>> locscale
> > >>>> motioncorr
> > >>>> phenix
> > >>>> resmap
> > >>>> sidesplitter
> > >>>> sphire
> > >>>> spider
> > >>>> topaz
> > >>>> chimera
> > >>>> eman2
> > python3
> /home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/apps/pw_run_tests.py
> eman2.tests.test_protocols_eman.TestEmanTiltValidate
> > Running tests....
> > Creating project at:
> /home/azazello/ScipionUserData/projects/TestEmanTiltValidate/project.sqlite
> > [ FAILED ] TestEmanTiltValidate.setUpClass
> >
> > Traceback (most recent call last):
> > File
> "/home/azazello/soft/miniconda3/envs/.scipion3env/lib/python3.8/unittest/suite.py",
> line 163, in _handleClassSetUp
> > setUpClass()
> > File
> "/home/azazello/soft/scipion-em-plugins/scipion-em-eman2/eman2/tests/test_protocols_eman.py",
> line 263, in setUpClass
> > cls.dataset = DataSet.getDataSet('eman')
> > File
> "/home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/tests/tests.py",
> line 57, in getDataSet
> > assert name in cls._datasetDict, "Dataset: %s dataset doesn't exist." %
> name
> > AssertionError: Dataset: eman dataset doesn't exist.
> >
> > Individual tests still work though.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/scipion-em/scipion-em/issues/200>, or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAF7ZYP4XI3L3LLJZUQHWQTTDTHAFANCNFSM4ZFCBUJA
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipion-em/scipion-em/issues/200#issuecomment-798958793>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBPJQE3TTGOZHXFYFNJ6LTDUAM5ANCNFSM4ZFCBUJA>
> .
>

You are right! the --run has a different behaviour. I need to look at 
this more in details. Thanks as always!

On 14/3/21 20:00, Grigory Sharov wrote:
> I have done git pull in all repos before creating this issue.
>
> Best regards,
> Grigory
>
> --------------------------------------------------------------------------------
> Grigory Sharov, Ph.D.
>
> MRC Laboratory of Molecular Biology,
> Francis Crick Avenue,
> Cambridge Biomedical Campus,
> Cambridge CB2 0QH, UK.
> tel. +44 (0) 1223 267228 <+44%201223%20267228>
> e-mail: ***@***.***
>
>
> On Sun, Mar 14, 2021 at 6:49 PM Pablo Conesa ***@***.***>
> wrote:
>
> > I should have warned you. Datasets were moved to scipion-em and we 
> need a
> > small change in scipion-app. Git pull of scipion app should fix it
> >
> > El dom., 14 mar. 2021 16:13, Grigory Sharov ***@***.***>
> > escribió:
> >
> > > After the sudden move of all tests datasets definition from pw to 
> pwem,
> > > the following command fails:
> > >
> > > ***@***.***:~/soft/scipion3$ scipion3 tests --grep eman --run
> > > Scipion v3.0.7 - Eugenius
> > > Running tests....
> > > >>>> xmipp3
> > > >>>> atlas
> > > >>>> bamfordlab
> > > >>>> bsoft
> > > >>>> ccp4
> > > >>>> cryoef
> > > >>>> cryodrgn
> > > >>>> emxlib
> > > >>>> fsc3d
> > > >>>> gautomatch
> > > >>>> imagic
> > > >>>> localrec
> > > >>>> locscale
> > > >>>> motioncorr
> > > >>>> phenix
> > > >>>> resmap
> > > >>>> sidesplitter
> > > >>>> sphire
> > > >>>> spider
> > > >>>> topaz
> > > >>>> chimera
> > > >>>> eman2
> > > python3
> > 
> /home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/apps/pw_run_tests.py
> > eman2.tests.test_protocols_eman.TestEmanTiltValidate
> > > Running tests....
> > > Creating project at:
> > 
> /home/azazello/ScipionUserData/projects/TestEmanTiltValidate/project.sqlite
> > > [ FAILED ] TestEmanTiltValidate.setUpClass
> > >
> > > Traceback (most recent call last):
> > > File
> > 
> "/home/azazello/soft/miniconda3/envs/.scipion3env/lib/python3.8/unittest/suite.py",
> > line 163, in _handleClassSetUp
> > > setUpClass()
> > > File
> > 
> "/home/azazello/soft/scipion-em-plugins/scipion-em-eman2/eman2/tests/test_protocols_eman.py",
> > line 263, in setUpClass
> > > cls.dataset = DataSet.getDataSet('eman')
> > > File
> > 
> "/home/azazello/soft/scipion3/scipion-pyworkflow/pyworkflow/tests/tests.py",
> > line 57, in getDataSet
> > > assert name in cls._datasetDict, "Dataset: %s dataset doesn't 
> exist." %
> > name
> > > AssertionError: Dataset: eman dataset doesn't exist.
> > >
> > > Individual tests still work though.
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/scipion-em/scipion-em/issues/200>, or unsubscribe
> > > <
> > 
> https://github.com/notifications/unsubscribe-auth/AAF7ZYP4XI3L3LLJZUQHWQTTDTHAFANCNFSM4ZFCBUJA
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/scipion-em/scipion-em/issues/200#issuecomment-798958793>,
> > or unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/ABVBPJQE3TTGOZHXFYFNJ6LTDUAM5ANCNFSM4ZFCBUJA>
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/scipion-em/scipion-em/issues/200#issuecomment-798960363>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAF7ZYK4YXFYYO7FHTCXLPDTDUBUVANCNFSM4ZFCBUJA>.
>

--- a/pwem/__init__.py
+++ b/pwem/__init__.py
@@ -37,6 +37,7 @@ import pyworkflow.plugin
 
 from .constants import *
 from .objects import EMObject
+from .tests import defineDatasets
 from .utils import *
 
 __version__ = '3.0.8'
@@ -116,6 +117,9 @@ class Plugin(pyworkflow.plugin.Plugin):
     def _defineVariables(cls):
         cls._defineVar(EM_ROOT_VAR, pwem.Config.EM_ROOT)
         cls._defineEmVar(MAXIT_HOME, 'maxit-10.1')
+
+        # Take this initialization event to define own datasets
+        defineDatasets()
 
     @classmethod
     def defineBinaries(cls, env):

--- a/pwem/tests/__init__.py
+++ b/pwem/tests/__init__.py
@@ -27,251 +27,253 @@
 # **************************************************************************
 from pyworkflow.tests import DataSet
 
-DataSet(name='igbmc_gempicker', folder='igbmc_gempicker',
-        files={
-            'allMics': 'micrographs/*.mrc',
-            'mic1': 'micrographs/KLH_Dataset_I_Test_0001.mrc',
-            'mic2': 'micrographs/KLH_Dataset_I_Test_0002.mrc',
-            'mask1': 'masks/mask_sideview.tif',
-            'mask2': 'masks/mask_topview.tif',
-            'templates': 'templates/*.mrc'})
+def defineDatasets():
+    """ Defines datasets for testing."""
+    DataSet(name='igbmc_gempicker', folder='igbmc_gempicker',
+            files={
+                'allMics': 'micrographs/*.mrc',
+                'mic1': 'micrographs/KLH_Dataset_I_Test_0001.mrc',
+                'mic2': 'micrographs/KLH_Dataset_I_Test_0002.mrc',
+                'mask1': 'masks/mask_sideview.tif',
+                'mask2': 'masks/mask_topview.tif',
+                'templates': 'templates/*.mrc'})
 
-DataSet(name='xmipp_tutorial', folder='xmipp_tutorial',
-        files={
-            'aligned_particles': 'gold/aligned_particles.sqlite',
-            'allMics': 'micrographs/*.mrc',
-            'boxingDir': 'pickingEman/info',
-            'boxingFile': 'pickingEman/info/BPV_1386_info.json',
-            'coordsGoldSqlite': 'gold/coordinates_gold.sqlite',
-            'ctfGold': 'gold/xmipp_ctf.ctfparam',
-            'ctf': 'ctfs/ctfs.sqlite',
-            'images10': 'gold/images10.xmd',
-            'mic1': 'micrographs/BPV_1386.mrc',
-            'mic2': 'micrographs/BPV_1387.mrc',
-            'mic3': 'micrographs/BPV_1388.mrc',
-            'micsGoldSqlite': 'gold/micrographs_gold.sqlite',
-            'micsGoldSqlite2': 'gold/micrographs2_gold.sqlite',
-            'micsGoldXmd': 'gold/micrographs_gold.xmd',
-            'micsSqlite': 'micrographs/micrographs.sqlite',
-            'particles': 'particles/*.hdf',
-            'particles1': 'particles/BPV_1386_ptcls.hdf',
-            'particles2': 'particles/BPV_particles.sqlite',
-            'particles3': 'particles/BPV_particles_aligned.sqlite',
-            'posAllDir': 'pickingXmipp/pickedAll',
-            'posSupervisedDir': 'pickingXmipp/pickedSupervised',
-            'rctCoords': 'rct/pickingXmipp',
-            'rctMicsT': 'rct/micrographs/*T.mrc',
-            'rctMicsU': 'rct/micrographs/*U.mrc',
-            'vol1': 'volumes/BPV_scale_filtered_windowed_64.vol',
-            'vol2': 'volumes/volume_1_iter_002.mrc',
-            'vol3': 'volumes/volume_2_iter_002.mrc',
-            'vol4': 'volumes/helix_59_4__6_7.vol',
-            'vol110': 'volumes/BPV_scale_filtered_windowed_110.vol',
-            'volumes': 'volumes/*.mrc',
-            'mask2d': 'masks/mask2d.xmp',
-            'mask3d': 'masks/mask.vol',
-            'ctfsDir': 'ctfs'})
+    DataSet(name='xmipp_tutorial', folder='xmipp_tutorial',
+            files={
+                'aligned_particles': 'gold/aligned_particles.sqlite',
+                'allMics': 'micrographs/*.mrc',
+                'boxingDir': 'pickingEman/info',
+                'boxingFile': 'pickingEman/info/BPV_1386_info.json',
+                'coordsGoldSqlite': 'gold/coordinates_gold.sqlite',
+                'ctfGold': 'gold/xmipp_ctf.ctfparam',
+                'ctf': 'ctfs/ctfs.sqlite',
+                'images10': 'gold/images10.xmd',
+                'mic1': 'micrographs/BPV_1386.mrc',
+                'mic2': 'micrographs/BPV_1387.mrc',
+                'mic3': 'micrographs/BPV_1388.mrc',
+                'micsGoldSqlite': 'gold/micrographs_gold.sqlite',
+                'micsGoldSqlite2': 'gold/micrographs2_gold.sqlite',
+                'micsGoldXmd': 'gold/micrographs_gold.xmd',
+                'micsSqlite': 'micrographs/micrographs.sqlite',
+                'particles': 'particles/*.hdf',
+                'particles1': 'particles/BPV_1386_ptcls.hdf',
+                'particles2': 'particles/BPV_particles.sqlite',
+                'particles3': 'particles/BPV_particles_aligned.sqlite',
+                'posAllDir': 'pickingXmipp/pickedAll',
+                'posSupervisedDir': 'pickingXmipp/pickedSupervised',
+                'rctCoords': 'rct/pickingXmipp',
+                'rctMicsT': 'rct/micrographs/*T.mrc',
+                'rctMicsU': 'rct/micrographs/*U.mrc',
+                'vol1': 'volumes/BPV_scale_filtered_windowed_64.vol',
+                'vol2': 'volumes/volume_1_iter_002.mrc',
+                'vol3': 'volumes/volume_2_iter_002.mrc',
+                'vol4': 'volumes/helix_59_4__6_7.vol',
+                'vol110': 'volumes/BPV_scale_filtered_windowed_110.vol',
+                'volumes': 'volumes/*.mrc',
+                'mask2d': 'masks/mask2d.xmp',
+                'mask3d': 'masks/mask.vol',
+                'ctfsDir': 'ctfs'})
 
-DataSet(name='mda', folder='hemoglobin_mda',
-        files={
-            'particles': 'particles/*.spi',
-            'particlesDir': 'particles',
-            'volumes': 'volumes/*.spi',
-            'volumesDir': 'volumes',
-            'averages': 'averages/averages.stk',
-            'averagesDir': 'averages'})
+    DataSet(name='mda', folder='hemoglobin_mda',
+            files={
+                'particles': 'particles/*.spi',
+                'particlesDir': 'particles',
+                'volumes': 'volumes/*.spi',
+                'volumesDir': 'volumes',
+                'averages': 'averages/averages.stk',
+                'averagesDir': 'averages'})
 
-DataSet(name='nma', folder='nma',
-        files={
-            'pdb': 'pdb/AK.pdb',
-            'vol': 'volumes/AK_LP10.vol',
-            'particles': 'particles/img.stk'})
+    DataSet(name='nma', folder='nma',
+            files={
+                'pdb': 'pdb/AK.pdb',
+                'vol': 'volumes/AK_LP10.vol',
+                'particles': 'particles/img.stk'})
 
-DataSet(name='powerfit', folder='powerfit',
-        files={
-            'pdb': '3j79-pdb-bundle1.pdb',
-            'vol': 'ribosome_4.vol'})
+    DataSet(name='powerfit', folder='powerfit',
+            files={
+                'pdb': '3j79-pdb-bundle1.pdb',
+                'vol': 'ribosome_4.vol'})
 
-DataSet(name='tomo', folder='xmipp_tomo_test',
-        files={
-            'vol1': 'volumes/subvols_6E6.001.mrc.spi',
-            'vol2': 'volumes/subvols_6E6.002.mrc.spi',
-            'vol3': 'volumes/subvols_6E6.003.mrc.spi',
-            'volumes': 'volumes/*.spi'})
+    DataSet(name='tomo', folder='xmipp_tomo_test',
+            files={
+                'vol1': 'volumes/subvols_6E6.001.mrc.spi',
+                'vol2': 'volumes/subvols_6E6.002.mrc.spi',
+                'vol3': 'volumes/subvols_6E6.003.mrc.spi',
+                'volumes': 'volumes/*.spi'})
 
-DataSet(name='relion_tutorial', folder='relion_tutorial',
-        files={
-            'allMics': 'micrographs/*.mrc',
-            'boxingDir': 'pickingEman',
-            'input_particles': 'gold/input_particles.star',
-            'particles': 'gold/particles.sqlite',
-            'posAllDir': 'pickingXmipp',
-            'relion_it020_data': 'gold/relion_it020_data.star',
-            'relion_it020_data_newlabels': 'gold/relion_it020_data_newlabels.star',
-            'volume': 'volumes/reference.mrc',
-            'import1_data_star': 'import/case1/classify3d_small_it038_data.star',
-            'import2_data_star': 'import/case2/relion_it015_data.star',
-            'particles_binary': '006.stk'
-        })
+    DataSet(name='relion_tutorial', folder='relion_tutorial',
+            files={
+                'allMics': 'micrographs/*.mrc',
+                'boxingDir': 'pickingEman',
+                'input_particles': 'gold/input_particles.star',
+                'particles': 'gold/particles.sqlite',
+                'posAllDir': 'pickingXmipp',
+                'relion_it020_data': 'gold/relion_it020_data.star',
+                'relion_it020_data_newlabels': 'gold/relion_it020_data_newlabels.star',
+                'volume': 'volumes/reference.mrc',
+                'import1_data_star': 'import/case1/classify3d_small_it038_data.star',
+                'import2_data_star': 'import/case2/relion_it015_data.star',
+                'particles_binary': '006.stk'
+            })
 
-DataSet(name='relion30_tutorial', folder='relion30_tutorial', files={})
+    DataSet(name='relion30_tutorial', folder='relion30_tutorial', files={})
 
-DataSet(name='ribo_movies', folder='ribo_movies',
-        files={
-            'movies': 'movies/1??_*.mrcs',
-            'posAllDir': 'pickingXmipp',
-            'volume': 'volumes/reference.mrc'})
+    DataSet(name='ribo_movies', folder='ribo_movies',
+            files={
+                'movies': 'movies/1??_*.mrcs',
+                'posAllDir': 'pickingXmipp',
+                'volume': 'volumes/reference.mrc'})
 
-DataSet(name='model', folder='model',
-        files={
-            'classesSelection': 'gold/classes_selection.sqlite',
-            'modelGoldSqlite': 'gold/model_gold.sqlite',
-            'modelGoldXml': 'gold/model_gold.xml'})
+    DataSet(name='model', folder='model',
+            files={
+                'classesSelection': 'gold/classes_selection.sqlite',
+                'modelGoldSqlite': 'gold/model_gold.sqlite',
+                'modelGoldXml': 'gold/model_gold.xml'})
 
-DataSet(name='rct', folder='rct',
-        files={
-            'classes': 'classes/classes2D_stable_core.sqlite',
-            'positions': 'positions',
-            'tilted': 'micrographs/F_rct_t*.tif',
-            'untilted': 'micrographs/F_rct_u*.tif'})
+    DataSet(name='rct', folder='rct',
+            files={
+                'classes': 'classes/classes2D_stable_core.sqlite',
+                'positions': 'positions',
+                'tilted': 'micrographs/F_rct_t*.tif',
+                'untilted': 'micrographs/F_rct_u*.tif'})
 
-DataSet(name='groel', folder='groel',
-        files={
-            'averages': 'classes/level_classes.stk',
-            'mic1': 'micrographs/groel-stain.mrc',
-            'pickingDogpicker': 'pickingDogpicker'})
+    DataSet(name='groel', folder='groel',
+            files={
+                'averages': 'classes/level_classes.stk',
+                'mic1': 'micrographs/groel-stain.mrc',
+                'pickingDogpicker': 'pickingDogpicker'})
 
-# TODO: Files to be deleted after checking no plugin or old scipion
-#  installations (1.2, ...) are using them:
-#  reconstRotandShiftFlip_Gold_output.vol (moved to reconstruction/gold)
-#  reconstRotandShift_Gold_output.vol (moved to reconstruction/gold)
-#  reconstRotOnly_Gold_output.vol (moved to reconstruction/gold)
+    # TODO: Files to be deleted after checking no plugin or old scipion
+    #  installations (1.2, ...) are using them:
+    #  reconstRotandShiftFlip_Gold_output.vol (moved to reconstruction/gold)
+    #  reconstRotandShift_Gold_output.vol (moved to reconstruction/gold)
+    #  reconstRotOnly_Gold_output.vol (moved to reconstruction/gold)
 
-DataSet(name='emx', folder='emx',
-        files={
-            'alignFlip': 'alignment/alignFlip.mrcs',
-            'alignFlip2': 'alignment/alignFlip2.mrcs',
-            'alignShiftRotExp': 'alignment/Test1/images.mrc',
-            'alignShiftRotExpGold': 'alignment/gold/align_shift_rot_exp_gold.mrcs',
+    DataSet(name='emx', folder='emx',
+            files={
+                'alignFlip': 'alignment/alignFlip.mrcs',
+                'alignFlip2': 'alignment/alignFlip2.mrcs',
+                'alignShiftRotExp': 'alignment/Test1/images.mrc',
+                'alignShiftRotExpGold': 'alignment/gold/align_shift_rot_exp_gold.mrcs',
 
-            'alignReconst': 'alignment/recons.mrcs',
-            'alignReconstShift': 'alignment/recons_sh.mrcs',
-            'alignRotOnly': 'alignment/align_rot_only.mrcs',
-            'alignRotOnlyGold': 'alignment/gold/align_rot_only_gold.mrcs',
-            # TODO: transform geometry not working in stacks of volumes
-            'alignRotOnly3D': 'alignment/align_rot_only_3d.mrcs',
-            # 'alignRotShift': 'alignment/align_rot_shift.mrcs',
-            'alignShiftOnly': 'alignment/align_shift_only.mrcs',
-            'alignShiftOnlyGold':
-                'alignment/gold/align_shift_only_gold.mrcs',
-            # TODO: transform geometry not working in stacks of volumes
-            'alignShiftOnly3D': 'alignment/align_shift_only_3d.mrcs',
-            'alignShiftRot': 'alignment/align_shift_rot.mrcs',
-            'alignShiftRotGold': 'alignment/gold/align_shift_rot_gold.mrcs',
-            # TODO: transform geometry not working in stacks of volumes
-            'alignShiftRot3D': 'alignment/align_shift_rot_3d.mrcs',
-            'coordinatesGoldT1':
-                'coordinates/Test1/coordinates_gold.sqlite',
-            'coordinatesT1': 'coordinates/Test1/coordinates.emx',
-            'defocusParticleT2': 'defocusParticle/particles.emx',
-            'emxMicrographCtf1': 'MicrographsCTF/ctfindCTFEstimation.emx',
-            'emxMicrographCtf1Gold':
-                'MicrographsCTF/ctfindCTFEstimation.sqlite',
-            'micrographsGoldT2': 'defocusParticle/micrographs_gold.sqlite',
-            'particlesGoldT2': 'defocusParticle/particles_gold.sqlite',
-            'reconstRotOnly': 'reconstruction/reconstRotOnly.mrcs',
-            'reconstRotOnlyGold':
-                'reconstruction/gold/reconstRotOnly_output.vol',
-            'reconstRotandShift': 'reconstruction/reconstRotandShift.mrcs',
-            'reconstRotandShiftGold':
-                'reconstruction/gold/reconstRotandShift_output.vol',
-            'reconstRotandShiftFlip':
-                'reconstruction/reconstRotandShiftFlip.mrcs',
-            'reconstRotandShiftFlipGold':
-                'reconstruction/gold/reconstRotandShiftFlip_output.vol',
-            'reconstRotOnlyGoldRln':
-                'reconstruction/gold/reconstRotOnly_rln_output.vol',
-            'reconstRotandShiftGoldRln':
-                'reconstruction/gold/reconstRotandShift_rln_output.vol',
-            'reconstRotandShiftFlipGoldRln':
-                'reconstruction/gold/reconstRotandShiftFlip_rln_output.vol'
-        })
+                'alignReconst': 'alignment/recons.mrcs',
+                'alignReconstShift': 'alignment/recons_sh.mrcs',
+                'alignRotOnly': 'alignment/align_rot_only.mrcs',
+                'alignRotOnlyGold': 'alignment/gold/align_rot_only_gold.mrcs',
+                # TODO: transform geometry not working in stacks of volumes
+                'alignRotOnly3D': 'alignment/align_rot_only_3d.mrcs',
+                # 'alignRotShift': 'alignment/align_rot_shift.mrcs',
+                'alignShiftOnly': 'alignment/align_shift_only.mrcs',
+                'alignShiftOnlyGold':
+                    'alignment/gold/align_shift_only_gold.mrcs',
+                # TODO: transform geometry not working in stacks of volumes
+                'alignShiftOnly3D': 'alignment/align_shift_only_3d.mrcs',
+                'alignShiftRot': 'alignment/align_shift_rot.mrcs',
+                'alignShiftRotGold': 'alignment/gold/align_shift_rot_gold.mrcs',
+                # TODO: transform geometry not working in stacks of volumes
+                'alignShiftRot3D': 'alignment/align_shift_rot_3d.mrcs',
+                'coordinatesGoldT1':
+                    'coordinates/Test1/coordinates_gold.sqlite',
+                'coordinatesT1': 'coordinates/Test1/coordinates.emx',
+                'defocusParticleT2': 'defocusParticle/particles.emx',
+                'emxMicrographCtf1': 'MicrographsCTF/ctfindCTFEstimation.emx',
+                'emxMicrographCtf1Gold':
+                    'MicrographsCTF/ctfindCTFEstimation.sqlite',
+                'micrographsGoldT2': 'defocusParticle/micrographs_gold.sqlite',
+                'particlesGoldT2': 'defocusParticle/particles_gold.sqlite',
+                'reconstRotOnly': 'reconstruction/reconstRotOnly.mrcs',
+                'reconstRotOnlyGold':
+                    'reconstruction/gold/reconstRotOnly_output.vol',
+                'reconstRotandShift': 'reconstruction/reconstRotandShift.mrcs',
+                'reconstRotandShiftGold':
+                    'reconstruction/gold/reconstRotandShift_output.vol',
+                'reconstRotandShiftFlip':
+                    'reconstruction/reconstRotandShiftFlip.mrcs',
+                'reconstRotandShiftFlipGold':
+                    'reconstruction/gold/reconstRotandShiftFlip_output.vol',
+                'reconstRotOnlyGoldRln':
+                    'reconstruction/gold/reconstRotOnly_rln_output.vol',
+                'reconstRotandShiftGoldRln':
+                    'reconstruction/gold/reconstRotandShift_rln_output.vol',
+                'reconstRotandShiftFlipGoldRln':
+                    'reconstruction/gold/reconstRotandShiftFlip_rln_output.vol'
+            })
 
-DataSet(name='CTFDiscrepancy', folder='CTFDiscrepancy',
-        files={
-            'emxMicrographCtf1': 'ctfindCTFEstimation.emx',
-            'emxMicrographCtf2': 'xmipp3CTFEstimation.emx',
-            'emxMicrographCtf3': 'xmipp3CTFEstimation2.emx',
-            'ctfsGold': 'ctfs.sqlite'})
+    DataSet(name='CTFDiscrepancy', folder='CTFDiscrepancy',
+            files={
+                'emxMicrographCtf1': 'ctfindCTFEstimation.emx',
+                'emxMicrographCtf2': 'xmipp3CTFEstimation.emx',
+                'emxMicrographCtf3': 'xmipp3CTFEstimation2.emx',
+                'ctfsGold': 'ctfs.sqlite'})
 
-DataSet(name='movies', folder='movies',
-        files={'movie1': 'ribo/Falcon_2012_06_12-14_33_35_0_movie.mrcs',
-               'movie2': 'ribo/Falcon_2012_06_12-16_55_40_0_movie.mrcs',
-               'movie3': 'ribo/Falcon_2012_06_12-17_26_54_0_movie.mrcs',
-               'movie4': 'qbeta/qbeta.mrc',
-               'movie5': 'cct/cct_1.em',
-               'falcon2012_1': 'Falcon_2012_06_12-16_55_40_0_movie.mrcs',
-               'mrcGain': 'gain.mrc',
-               'mrcDark': 'dark.mrc',
-               'spiGain': 'gain.spi',
-               'spiDark': 'dark.spi',
-               'eer': 'FoilHole_29355520_Data_29330528_29330530_20200329_234551_Fractions.mrc.eer'
-               })
-DataSet(name='riboMovies', folder='riboMovies',
-        files={'movie1': '30_movie_gc_window.mrcs',
-               'movie2': '35_movie_gc_window.mrcs',
-               'movie3': '130_movie_gc_window.mrcs',
-               'movie4': '165_movie_gc_window.mrcs',
-               'movie5': '270_movie_gc_window.mrcs',
-               'allMovies': '*.mrcs'})
+    DataSet(name='movies', folder='movies',
+            files={'movie1': 'ribo/Falcon_2012_06_12-14_33_35_0_movie.mrcs',
+                   'movie2': 'ribo/Falcon_2012_06_12-16_55_40_0_movie.mrcs',
+                   'movie3': 'ribo/Falcon_2012_06_12-17_26_54_0_movie.mrcs',
+                   'movie4': 'qbeta/qbeta.mrc',
+                   'movie5': 'cct/cct_1.em',
+                   'falcon2012_1': 'Falcon_2012_06_12-16_55_40_0_movie.mrcs',
+                   'mrcGain': 'gain.mrc',
+                   'mrcDark': 'dark.mrc',
+                   'spiGain': 'gain.spi',
+                   'spiDark': 'dark.spi',
+                   'eer': 'FoilHole_29355520_Data_29330528_29330530_20200329_234551_Fractions.mrc.eer'
+                   })
+    DataSet(name='riboMovies', folder='riboMovies',
+            files={'movie1': '30_movie_gc_window.mrcs',
+                   'movie2': '35_movie_gc_window.mrcs',
+                   'movie3': '130_movie_gc_window.mrcs',
+                   'movie4': '165_movie_gc_window.mrcs',
+                   'movie5': '270_movie_gc_window.mrcs',
+                   'allMovies': '*.mrcs'})
 
-DataSet(name='jmbFalconMovies', folder='jmbFalconMovies',
-        files={'movie1': 'Falcon_2014_07_16-16_32_43_0.mrcs',
-               'movie2': 'Falcon_2014_07_16-21_37_37_0.mrcs',
-               'movie3': 'Falcon_2014_07_16-23_30_50_0.mrcs',
-               'movie4': 'Falcon_2014_07_17-03_21_57_0.mrcs',
-               'movie5': 'Falcon_2014_07_17-06_55_56_0.mrcs',
-               'allMovies': '*.mrcs'})
+    DataSet(name='jmbFalconMovies', folder='jmbFalconMovies',
+            files={'movie1': 'Falcon_2014_07_16-16_32_43_0.mrcs',
+                   'movie2': 'Falcon_2014_07_16-21_37_37_0.mrcs',
+                   'movie3': 'Falcon_2014_07_16-23_30_50_0.mrcs',
+                   'movie4': 'Falcon_2014_07_17-03_21_57_0.mrcs',
+                   'movie5': 'Falcon_2014_07_17-06_55_56_0.mrcs',
+                   'allMovies': '*.mrcs'})
 
-DataSet(name='grigorieff', folder='grigorieff',
-        files={'particles_rot_shift': 'particles.sqlite',
-               'ref_volume': 'ref_volume.vol',
-               'ctfsDir': 'ctfs'})
+    DataSet(name='grigorieff', folder='grigorieff',
+            files={'particles_rot_shift': 'particles.sqlite',
+                   'ref_volume': 'ref_volume.vol',
+                   'ctfsDir': 'ctfs'})
 
-DataSet(name='initial_volume', folder='initial_volume',
-        files={'groel': 'groel_64.stk',
-               'ribosome': 'ribosomeFrank_64_preprocessed.stk',
-               'bpv': 'bpv_100.stk'})
+    DataSet(name='initial_volume', folder='initial_volume',
+            files={'groel': 'groel_64.stk',
+                   'ribosome': 'ribosomeFrank_64_preprocessed.stk',
+                   'bpv': 'bpv_100.stk'})
 
-DataSet(name='general', folder='general',
-        files={'vol_helix': 'volumes/helix_59_4__6_7.vol'})
+    DataSet(name='general', folder='general',
+            files={'vol_helix': 'volumes/helix_59_4__6_7.vol'})
 
-DataSet(name='SymVirus', folder='SymVirus',
-        files={'whole_vol_half': 'whole_vol_half.vol',
-               'large_vol_half_th': 'large_vol_half_th.vol',
-               'small_vol_half_th': 'small_vol_half_th.vol'})
+    DataSet(name='SymVirus', folder='SymVirus',
+            files={'whole_vol_half': 'whole_vol_half.vol',
+                   'large_vol_half_th': 'large_vol_half_th.vol',
+                   'small_vol_half_th': 'small_vol_half_th.vol'})
 
-DataSet(name='xmipp_programs', folder='xmipp_programs', files={})
+    DataSet(name='xmipp_programs', folder='xmipp_programs', files={})
 
-DataSet(name='resmap', folder='resmap',
-        files={'betagal': 'betaGal.mrc',
-               'betagal_half1': 'betaGal_half01.mrc',
-               'betagal_half2': 'betaGal_half02.mrc',
-               'betagal_mask': 'betaGal_mask.mrc'})
+    DataSet(name='resmap', folder='resmap',
+            files={'betagal': 'betaGal.mrc',
+                   'betagal_half1': 'betaGal_half01.mrc',
+                   'betagal_half2': 'betaGal_half02.mrc',
+                   'betagal_mask': 'betaGal_mask.mrc'})
 
-DataSet(name='10010', folder='10010',
-        files={'initialVolume': 'initialVolume.vol',
-               'particles': 'particles.sqlite'})
+    DataSet(name='10010', folder='10010',
+            files={'initialVolume': 'initialVolume.vol',
+                   'particles': 'particles.sqlite'})
 
-DataSet(name='model_building_tutorial', folder='model_building_tutorial',
-        files={'vol1': 'volumes/1ake_4-5A.mrc',
-               'pdb1': 'PDBs/1ake_start.pdb'})
+    DataSet(name='model_building_tutorial', folder='model_building_tutorial',
+            files={'vol1': 'volumes/1ake_4-5A.mrc',
+                   'pdb1': 'PDBs/1ake_start.pdb'})
 
-DataSet(name='eman', folder='eman',
-        files={
-            'micU': 'mics/ip3r10252011-0005_0-2.hdf',
-            'micT': 'mics/ip3r10252011-0005_10.hdf',
-            'volume': '3DmapIP3R1_clip_376apix.mrc',
-            'coords': 'coords/*.json',
-            'ctfs': 'ctfs/'
-        })
+    DataSet(name='eman', folder='eman',
+            files={
+                'micU': 'mics/ip3r10252011-0005_0-2.hdf',
+                'micT': 'mics/ip3r10252011-0005_10.hdf',
+                'volume': '3DmapIP3R1_clip_376apix.mrc',
+                'coords': 'coords/*.json',
+                'ctfs': 'ctfs/'
+            })

--- a/pwem/tests/__init__.py
+++ b/pwem/tests/__init__.py
@@ -277,3 +277,9 @@ def defineDatasets():
                 'coords': 'coords/*.json',
                 'ctfs': 'ctfs/'
             })
+
+    DataSet(name='deepConsensusPicking', folder='deepConsensusPicking',
+            files={
+                'allMics': 'micrographs/*.mrc',
+                'autoPickingCoordinates': 'autoPickingCoordinates'
+            })

